docs/articles/baseplatform/reproducible-yocto-builds.rst

```diff
@@ -0,0 +1,134 @@
+
+Building with BB_NO_NETWORK
+---------------------------
+
+Setting the ``BB_NO_NETWORK`` variable to "1" in build/conf/local.conf disables network access for
+bitbake during the build so that source code can only be fetched from specified local mirrors.
+This is a good way to guarantee that only specific sources will be used. However, this also means
+that bitbake will fail when issuing certain commands, for instance git ls-remote, and that it is
+therefore important to use the ``SRC_REV`` variable instead of pointing to a tag in git-based
+bitbake recipes. A typical failure could look like this:
+
+.. code-block:: none
+
+    Function failed: Network access disabled through BB_NO_NETWORK (or set indirectly due to use of BB_FETCH_PREMIRRORONLY) but access requested with command git ls-remote http://git.projects.genivi.org/ipc/common-api-dbus-runtime.git refs/heads/2.1.6 refs/tags/2.1.6^{} (for url None)
+    ERROR: Task 715 (/mnt/ssd/yocto/the-reproduced-build/sources/meta-pelagicore/recipes-core/common-api/common-api-c++-dbus_2.1.6.bb, do_fetch) failed with exit code '1'
+
+The poky layers do not allow recipes that cannot be built with ``BB_NO_NETWORK`` set to "1",
+but other upstream layers might. If such a recipe is found in some other bitbake layer repository,
+a bbappend file overriding the ``SRC_URL`` variable can be created so that the recipe instead will
+point to a specific commit.
+
+
+Configuring Yocto builds for reproducibility
+--------------------------------------------
+
+To create a local mirror, the following configuration needs to be added to build/conf/local.conf
+(note that this is not a default configuration):
+
+.. code-block:: none
+
+    BB_GENERATE_MIRROR_TARBALLS = "1"
+
+During the Yocto build process, packages are downloaded and stored in the downloads/ directory
+(or whatever directory has been specified in build/conf/local.conf) of the Yocto build's
+root directory.
+When set to "1", ``BB_GENERATE_MIRROR_TARBALLS`` forces bitbake to create tar-balls for all the
+sources that are fetched using git.
+Note that the build must be done from scratch otherwise some source code packages may not be
+packaged into tar-balls. In other words, the build cache needs to be wiped clean before the build
+is started. When everything is set up, start the build as usual with bitbake.
+
+
+Saving the repo manifest
+------------------------
+
+The above instructions describe how to create the local mirror.To be able to reproduce a build,
+the manifest file used must also be saved.
+The manifest with the exact commits used needs to be generated.
+This can be done with the repo tool from the root of the Yocto build directory after the repo sync
+command has been issued:
+
+.. code-block:: bash
+
+    $ repo manifest -o - -r >> example-manifest.xml
+
+Reproducing the Yocto build
+---------------------------
+
+The instructions in this section describes how a Yocto build may be reproduced given that the
```

Comment:
Still unfixed:
"describes" -> "describe"

"given" -> "assuming"

Reply:
Agree on "describes" -> "describe"
don't agree on "given" -> "assuming"

Reply:
"given" is actually wrong here, because it means that the fact which follows is taken for certain (see also http://www.oxfordlearnersdictionaries.com/definition/english/given-that).
If you don't like "assuming that", a probably better option is "provided that".

Some more opinions: https://ell.stackexchange.com/a/34636